subsys/bluetooth/audio/ascs.c

```diff
@@ -812,17 +880,27 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 	}
 
 	for (size_t i = 0; i < ARRAY_SIZE(session->ases); i++) {
-		struct bt_ascs_ase *ase = &session->ases[i];
-		struct bt_audio_stream *stream = ase->ep.stream;
+		if (session->ases[i]) {
```

Comment:
Is there any case where `ases[0]` may be NULL and `ases[1]` isn't? In other words, can we break this loop when we meet a NULL `session->ases[i]`, or do we need to iterate on all of them?

Reply:
Actually we do. If at some point two ASEs are allocated for a connection, index  0 and 1, then only index 0 was deallocated through a Release operation we would still have to handle index 1.

Maybe this also speaks for a sys_slist solution instead, as was commented earlier I saw :)